### PR TITLE
ecip-1105: becomes Ancestor Hash Appended proposal

### PR DESCRIPTION
**Pull request by meowsbits**
This strips off all proposed context types and their
generic structure to instead propose only adding
a field 'ancestorHash' which defines a dependency
on some foregoing canonical block for a transaction.

### CODE REVIEW

File: _specs/ecip-1105.md [hunk 1]
```diff
@@ -52,177 +43,93 @@ Generally, these constraints give the consumer (the transactor) an ability to ex
...
+#### `ancestorHash`
 
-#### `eligibleMinerList`
+- `ancestorHash` `bytes`. A byte array 32 bytes in length.
 
-- `ANNOTATION_PREFIX` `2`.
-- `ELIGIBLE_MINER_LIST` `[address...]`. A list of addresses.
-- `MAX_ELEMENTS` `3`. The maximum number of addresses that can be provided.
+The `ancestorHash` is a reference to a specific block by hash. 
+If a transaction filling this field is included in block number `n`, the annotated value must match some block numbered `<= n-1` and existing as a direct (canonical; non-ommer) ancestor of the transaction's containing block.
```
Comment (BelfordZ) [inline _specs/ecip-1105.md:63]:
```suggestion    
 T1 ∈ B ⇔ BlockNumber(B) <= BlockNumber(RH)
  
 for transaction T1 with referenceHash RH,  mined block B
```

**Comment by BelfordZ** [comment]
nice one!

Might be nice to make it a `blockHash | transactionHash`, allowing similar thing but for creating transaction ordering dependencies. 

If enough transactions set this field, any chain mined without the required ancestorblock would have a very low cumulative difficulty unless they produced a much larger number of blocks than the cannonical chain. I'd imagine you would see overall far fewer less reorgs.

**Comment by gitr0n1n** [comment]
@meowsbits , if you can resolve @BelfordZ 's comments (if anything needs resolving), I'll allocate the time for a review within the week.

File: _specs/ecip-1105.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (meowsbits) [reply _specs/ecip-1105.md:63]: Actually wouldn't it be `T ∈ B ⇔ BlockNumber(B) >= BlockNumber(RH)`, since the block containing the subject transaction T will be a greater number than the block cited by the transaction as an `RH`. 

Also this logic-style syntax doesn't include the canonical-ancestry caveat. 

I'm working on a more academic-style paper for this idea that'll have these kinds of terms. For the ECIP, I'm OK leaving it (mostly) in plain English :).

**Comment by gitr0n1n** [comment]
> @BelfordZ
> 
> > Might be nice to make it a blockHash | transactionHash, allowing similar thing but for creating transaction ordering dependencies.
> 
> Yea, transaction-ordering seems like a logical next step (or adjacent abstraction), but...
> 
> a.) With only block hash dependencies we get a kind of superset of transaction-ordering dependency, since a block's hash will change if its transactions are reordered (or otherwise edited)... the block becomes a new block. So a transaction's citation of a block already cites the existence and ordering of its transactions.
> 
> b.) Transaction dependencies (vs. block dependencies) are generally less "strict." This feature is proposed in the context of [1108](https://github.com/ethereumclassic/ECIPs/blob/master/_specs/ecip-1108.md), which is interested in differentiating public vs. private chains, generally to mitigate the risk of large reorgs by increasing the cost of making them in independent isolation. An attacker in these circumstances can include any necessary transactions in their private chain more easily than blocks, which are impossible to reproduce.
> 
> > If enough transactions set this field, any chain mined without the required ancestorblock would have a very low cumulative difficulty unless they produced a much larger number of blocks than the cannonical chain. I'd imagine you would see overall far fewer less reorgs.
> 
> As for the reorgs, I'm not 100% sure I follow here (at least in the context of this ECIP). But in the context of 1108, indeed, the more transactions that use relevant (~contemporary) block dependencies, the fewer transactions will be applicable to private chains, and the more balance/capital the attacker (building a private chain) will need to provide themselves to stay competitive with the public aggregate transaction-sender balance score (TABS).
> 
> Will there be fewer (small) reorgs? Probably, but as I had understood it, that'll be because the TABS score (the 2nd-dimension consensus score value) for two competitive blocks (same parent), one of which is destined to become an ommer, the TABS score could be decisive around 50% of the time, breaking the consensus tie that otherwise difficulty alone would yield to a coin toss.

Great, I'm going to process the ECIP editor review. It appears `copy` for the ECIp is being edited here and perhaps the discussion @BelfordZ and @meowsbits are having can continue in the `Discussions-To` thread, as I think its great thoughtful technical conversation that would be great to have a clear paper trail for future readers. :+1: https://github.com/ethereumclassic/ECIPs/issues/422

**Review by gitr0n1n — APPROVED** [review]
Reviewed.

Appears to update ECIP for more clarity of proposal. removing unnecessary/not applicable copy. Nice read. Interesting proposal.
